### PR TITLE
Breadcrumbs dropdown

### PR DESCRIPTION
**Pull request by EmmanuelVella**
This PR adds dropdowns in breadcrumbs, so each breadcrumb element gets (if necessary) a dropdown which contains its corresponding action links, so it's easier to find the one you need. 

breadcrumbs-dropdown

### CODE REVIEW

**Comment by OskarStark** [comment]
I really like this PR!

Could you please add some tests? and have a look at travis

File: Admin/Admin.php [hunk 1]
```diff
@@ -2116,49 +2107,113 @@ public function buildBreadcrumbs($action, MenuItemInterface $menu = null)
...
 
         return $this->breadcrumbs[$action] = $menu;
     }
 
+    private function addBreadcrumbsDropdown(MenuItemInterface $menu, $subject = null)
+    {
+        if ($subject) {
+            $id = $this->request->get($this->getIdParameter());
+
+            $links = array(
```
Comment (OskarStark) [inline Admin/Admin.php]: these links are already defined anywhere... we should use them from there, so we don't get different action dropdowns

**Comment by EmmanuelVella** [comment]
Sure I will continue my work and add some tests if there is a good feedback about the PR ;)

**Comment by Soullivaneuh** [comment]
Sounds exiting! :+1:

File: Resources/views/standard_layout.html.twig [hunk 2]
```diff
@@ -263,23 +275,6 @@ file that was distributed with this source code.
                                                         {% endfor %}
                                                     </div>
                                                 {% endif %}
-                                                
-                                                {% block sonata_admin_content_actions_wrappers %}
```
Comment (rande) [inline Resources/views/standard_layout.html.twig]: We still need to support this block. I think we must should keep it.

File: Admin/Admin.php (hunk 1 above)
Comment (EmmanuelVella) [inline Admin/Admin.php]: You talk about the twig templates ?

File: Resources/views/standard_layout.html.twig (hunk 2 above)
Comment (EmmanuelVella) [inline Resources/views/standard_layout.html.twig]: Fixed

**Comment by Soullivaneuh** [comment]
Tests need to be fixed.

BTW, tests need to be added for this new feature.

**Comment by EmmanuelVella** [comment]
I have updated my PR :

**Comment by EmmanuelVella** [comment]
I just added a demo gif, tests and doc to my PR.

Here are some details / questions : 
1) I added a `configureBreadcrumbs` method in the admin / admin extensions. I finally don't think using the `configureTabMenu` is a good idea, especially because they may contain dropdowns (and only 1 level is currently possible).
2) I restored the top bar and the action links which are defined in a different way (templates). I don't know how to handle this.

**Comment by EmmanuelVella** [comment]
I don't understand what is wrong with the tests, could someone tell me ?

**Comment by Soullivaneuh** [comment]
@EmmanuelVella Tests are ok according to Travis.

Ignore `SonataCi / Sandbox / PHPUnit`, seems to be buggy from a time (cc @rande).

I'll checkout your branch for test later. :+1:

**Comment by EmmanuelVella** [comment]
@Soullivaneuh Ok great ! Thank you.

**Comment by rande** [comment]
@Soullivaneuh it is not buggy: 

```
Running tests from ./vendor/sonata-project/timeline-bundle
--
PHPUnit 4.8.9 by Sebastian Bergmann and contributors.

.PHP Fatal error:  Class Mock_AdminInterface_89e80999 contains 135 abstract methods and must therefore be declared abstract or implement the remaining methods (Sonata\AdminBundle\Admin\AdminInterface::setFormContractor, Sonata\AdminBundle\Admin\AdminInterface::setListBuilder, Sonata\AdminBundle\Admin\AdminInterface::getListBuilder, ...) in /var/www/vendor/phpunit/phpunit-mock-objects/src/Framework/MockObject/Generator.php(290) : eval()'d code on line 159
..Unit Test Suite has failed: vendor/sonata-project/timeline-bundle
================================================================================
```

So it is not related to this PR. But still there is an issue with unit test on the TimelineBundle.

**Comment by Soullivaneuh** [comment]
> So it is not related to this PR. But still there is an issue with unit test on the TimelineBundle.

Yeah, this is what I mean by "buggy". :-)

BTW, having a direct link to test results on PR would be much better. :+1:

**Comment by rande** [comment]
@Soullivaneuh yep, need to secure this part a bit more ;)

**Comment by Soullivaneuh** [comment]
@EmmanuelVella looks great, excepts two things for me:

* Menu should not appear on last breadcrumb element. It already exists on the page.
* Menu and breadcrumb element should be synchronized with the page menu itself. Actually we have multiple and different menu. If we want to customize, we have to to twice.

**Comment by EmmanuelVella** [comment]
@Soullivaneuh : 
- Ok, that's not a problem
- Do you have an idea how to achieve that ?

**Comment by Soullivaneuh** [comment]
> Do you have an idea how to achieve that ?

At this instant no. I let you search onto it. ;-)

**Comment by EmmanuelVella** [comment]
@Soullivaneuh we could rename the `configureBreadcrumbsMenu` method to`configureActionsMenu`, and then use this method to add dropdowns to the breadcrumb and display the action menu.

And the only way I see to keep BC for the "old" action templates. is to display a second "legacy" menu (if not empty) next the the new one.

**Comment by OskarStark** [comment]
@EmmanuelVella can you please rebase your PR?

@Soullivaneuh can you please give us feedback on this PR?

**Comment by EmmanuelVella** [comment]
@OskarStark Rebase is ok

I need a feedback about my previous comment to complete this PR.

**Comment by OskarStark** [comment]
ping @Soullivaneuh

**Comment by Soullivaneuh** [comment]
> @Soullivaneuh we could rename the configureBreadcrumbsMenu method toconfigureActionsMenu, and then use this method to add dropdowns to the breadcrumb and display the action menu.

I would prefer something like that.

**Comment by core23** [comment]
Any news on this? @EmmanuelVella

**Comment by greg0ire** [comment]
ping @EmmanuelVella

**Comment by greg0ire** [comment]
No news for a long time… can we remove this from the 2.4 milestone ?

File: Admin/AdminExtensionInterface.php [hunk 3]
```diff
@@ -54,6 +54,16 @@ public function configureShowFields(ShowMapper $show);
...
 
     /**
+     * @param AdminInterface    $admin
+     * @param MenuItemInterface $menu
+     * @param string            $action
+     * @param AdminInterface    $childAdmin
+     *
+     * @return mixed
+     */
+    public function configureBreadcrumbsMenu(AdminInterface $admin, MenuItemInterface $menu, $action, AdminInterface $childAdmin = null);
```
Comment (core23) [inline Admin/AdminExtensionInterface.php]: BC break. Can you comment this and create a new PR for the next major release?

**Comment by core23** [comment]
Ping @EmmanuelVella

**Comment by EmmanuelVella** [comment]
Sorry, I didn't receive github notifications since a while, just realized it. I will have a look in the next days !

**Comment by Soullivaneuh** [comment]
According to the new [Sonata version management](https://github.com/sonata-project/SonataAdminBundle/issues/3053) and next major [release plan](https://github.com/sonata-project/SonataAdminBundle/issues/3731), this project has been refactored regarding branching and versioning.

If you see this message, your PR concerns a patch or a minor release and is not targeting the right branch.

So I'm closing this one, but don't see it as a refusal. If you think your work is still relevant and want to continue, feel free to reopen it on the right branch (e.g. the default one).

Regards.

**Comment by core23** [comment]
@EmmanuelVella Are you still working on a new PR for this?

**Comment by EmmanuelVella** [comment]
@core23 Not at the moment, sorry !

**Comment by core23** [comment]
Is this code working or anything to do from your side @EmmanuelVella ? So I can simple rebase your PR?

**Comment by EmmanuelVella** [comment]
Yes it's working (would need a rebase), however like @Soullivaneuh said, breadcrumbs dropdown and page menu (to right) use different code, which implies a double work (see https://github.com/sonata-project/SonataAdminBundle/pull/3258#issuecomment-143742309)

**Pull request closed without merging**